src/Calypso-SystemPlugins-Reflectivity-Browser/ClyMetaLinkInstallationPresenter.class.st

```diff
@@ -0,0 +1,123 @@
+Class {
+	#name : #ClyMetaLinkInstallationPresenter,
+	#superclass : #SpPresenter,
+	#instVars : [
+		'sourceNode',
+		'list',
+		'toolBar',
+		'isForMetaLinkInstallation'
+	],
+	#category : #'Calypso-SystemPlugins-Reflectivity-Browser-Metalinks'
+}
+
+{ #category : #specs }
+ClyMetaLinkInstallationPresenter class >> defaultSpec [
+	^ SpBoxLayout newVertical
+		  add: #toolBar
+		  expand: false
+		  fill: false
+		  padding: 0;
+		  add: #list;
+		  yourself
+]
+
+{ #category : #'instance creation' }
+ClyMetaLinkInstallationPresenter class >> onNode: aSourceNode forInstallation: forInstallation [
+	^ self basicNew
+		  sourceNode: aSourceNode;
+		  isForMetaLinkInstallation: forInstallation;
+		  initialize
+]
+
+{ #category : #opening }
+ClyMetaLinkInstallationPresenter class >> openInstallerOnNode: aSourceNode [
+	(self onNode: aSourceNode forInstallation: true) openWithSpec
+]
+
+{ #category : #opening }
+ClyMetaLinkInstallationPresenter class >> openUninstallerOnNode: aSourceNode [
+	(self onNode: aSourceNode forInstallation: false) openWithSpec
+]
+
+{ #category : #accessing }
+ClyMetaLinkInstallationPresenter >> allRelevantMetaLinks [
+	| baseLinks |
+	baseLinks := self isForMetaLinkInstallation
+		             ifTrue: [ MetaLink allInstances ]
+		             ifFalse: [ sourceNode links ].
+	^ baseLinks reject: [ :link | 
+		  link metaObject == Break or: [ 
+			  { 
+				  Watchpoint.
+				  ExecutionCounter } includes: link metaObject class ] ]
```

Comment:
So the idea here is to remove from the list Break, Watchpoint and ExecutionCounter?

Reply:
Yes, because those are not metalinks but tools using them.
We should not interfere.
The best would be an full configuration UI that would allow, for each selected node, to configure all possible options, we are discussing that with @dupriezt.
I imagine something similar (but not the same) as intelliJ or Eclipse breakpoint window.